pkg/rollout/rollout.go

```diff
@@ -148,17 +148,18 @@ func (r *Rollout) UpdateService(svc *run.Service) (*run.Service, error) {
 		return nil, errors.Wrapf(err, "failed to diagnose health for candidate %q", candidate)
 	}
 
-	svc, err = r.updateServiceBasedOnDiagnosis(svc, diagnosis.OverallResult, stable, candidate)
+	traffic, err := r.trafficBasedOnDiagnosis(svc, diagnosis.OverallResult, stable, candidate)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to update service after diagnosis")
+		return nil, errors.Wrap(err, "failed to configure traffic after diagnosis")
 	}
-	if svc == nil {
-		// If service was unchanged, nil is returned.
```

Comment:
I left it as nil instead of returning the same, unchanged traffic. This is because we need to differentiate between an unchanged traffic because of inconclusive/no enough time since last rollout. So, it is easy to check for that if we get a `nil` traffic.

I think we could also use `go-cmp` to compare the old traffic to the new one to determine if the traffic was unchanged.